src/ride/track.c

```diff
@@ -4921,15 +4922,492 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+			continue;
+		}
+		if (mapElement->properties.track.type != 0x65) {
+			continue;
+		}
+
+		return loc_6CDB35(x, flags, direction, y, rideIndex, mode, unknown, z, mapElement);
+	} while (!map_element_is_last_for_tile(mapElement++));
+
+
+	if (mode != 0) {
+		// loc_6CDE43
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = 0;
+		// loc_6CDE4C
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	if (!map_can_construct_at((uint16) floor2(x, 32), (uint16) floor2(y, 32), (z / 8), (z / 8) + 4, 0x0F)) {
+		// loc_6CDE4C
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	if (RCT2_GLOBAL(RCT2_ADDRESS_ELEMENT_LOCATION_COMPARED_TO_GROUND_AND_WATER, uint8) & 0x04) {
+		// loc_6CDE22
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = STR_RIDE_CANT_BUILD_THIS_UNDERWATER;
+		// loc_6CDE4C
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	if (RCT2_GLOBAL(RCT2_ADDRESS_ELEMENT_LOCATION_COMPARED_TO_GROUND_AND_WATER, uint8) & 0x02) {
+		// loc_6CDE17
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = STR_CAN_ONLY_BUILD_THIS_ABOVE_GROUND;
+		// loc_6CDE4C
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	rct_ride *ride = get_ride(rideIndex);
+
+	money32 price = (((RCT2_ADDRESS(0x0097DD78, money16)[ride->type * 2] * RCT2_GLOBAL(0x0099DBC8, money32)) >> 16));
+	RCT2_GLOBAL(0x00F4413E, money32) = price / 2 * 10;
+
+
+	if ((flags & GAME_COMMAND_FLAG_APPLY) == 0) {
+		return loc_6CDDFD();
+	}
+
+
+	// TODO: shouldn't these arguments get shifted?
+	uint16 flooredX = floor2(x, 32);
+	uint16 flooredY = floor2(y, 32);
+
+	mapElement = map_element_insert(flooredX / 32, flooredY / 32, z >> 3, 0xF);
+	mapElement->clearance_height = (z >> 3) + 4;
+	mapElement->type = 0x8;
+	mapElement->properties.track.type = 0x65;
+	mapElement->properties.track.ride_index = rideIndex;
+	mapElement->properties.track.maze_entry = 0xFFFF;
+
+	if (flags & GAME_COMMAND_FLAG_GHOST) {
+		mapElement->flags |= MAP_ELEMENT_FLAG_GHOST;
+	}
+
+	// loc_6CDAE8:
+	map_invalidate_tile_full(flooredX, flooredY);
+
+	ride->maze_tiles += 1;
+	ride->station_heights[0] = mapElement->base_height;
+	ride->station_starts[0] = 0;
+
+	if (direction == 4) {
+		if (flags & GAME_COMMAND_FLAG_GHOST) {
+			ride->overall_view = (uint16) (((flooredX & 0xFFFF) >> 5) & ((flooredY << 3) & 0xFFFF));
+		}
+	}
+
+	return loc_6CDB35(x, flags, direction, y, rideIndex, mode, unknown, z, mapElement);
+}
+
 /**
  *
  *  rct2: 0x006CD8CE
  */
 void game_command_set_maze_track(int *eax, int *ebx, int *ecx, int *edx, int *esi, int *edi, int *ebp)
 {
-	RCT2_CALLFUNC_X(0x006CD8CE, eax, ebx, ecx, edx, esi, edi, ebp);
+	uint16 x = (uint16) (*eax & 0xFFFF); // AX
+	uint8 flags = (uint8) (*ebx & 0xFF); // BL
+	uint8 direction = (uint8) ((*ebx & 0xFF00) >> 8); // BH
+	uint16 y = (uint16) (*ecx & 0xFFFF); // CX
+	uint8 rideIndex = (uint8) (*edx & 0xFF); // DL
+	uint8 mode = (uint8) ((*edx & 0xFF00) >> 8); // DH
+	uint16 unknown = (uint16) ((*edx & 0xFFFF0000) >> 16); // EDX
```

Comment:
@duncanspumpkin I've been checking in IDA, but I can't find any place where this value might get used.